src/base/async/int.c

```diff
@@ -2724,3 +2722,90 @@ void update_xtitle(void)
 	}
     }
 }
+
+static int msdos_remap_extended_open_(void)
+{
+  uint16_t _si_ = LWORD(esi);
+  uint8_t _bl_ = LO(bx);
+  uint8_t _dl_ = LO(dx);
+  char *src = MK_FP32(SREG(ds), _si_);
+  int found, create_truncate;
+
+  ds_printf("INT21: extended open '%s'\n", src);
+
+  /* Get file attributes */
+  LWORD(eax) = 0x4300;
+  LWORD(edx) = _si_;                           // Filename passed in DS:DX
+  call_msdos();
+  if (!(REG(eflags) & CF))
+    LWORD(eax) = 0x00;
```

Comment:
Missing else?

Reply:
Btw, there are `pre_msdos()` and `post_msdos()`
helpers that do nothing but save/restore registers.
So that you don't need to save copies in `_dl_` etc.

Reply:
Nope, if the carry flag is set then so is the AX register.

Reply:
If missing else is intentional, please add
comment why to continue with CF and
what will be in AX.

Reply:
> Nope, if the carry flag is set then so is the AX register.

To what value? And why is it interesting after
CF is set?

Reply:
If CF set then AX = doserror code
if CF not set then the getattr succeeded(file must exist) AX is undefined, so I set it to 0.

The only two states I wish to continue with are; file exists; and file doesn't. Anything else exit early with the doserror.

Reply:
Then perhaps much more readable would be:
```
if (isset_CF() && _AX != 2)
    return 0;
found = !isset_CF();
```
Fixing up dos register (set AX to 0) only to check
it later is a bad coding style - you use it as a local variable.
Makes code less readable and disables some gcc
optimizations.

Reply:
```C
if (isset_CF() && _AX != 2)
  return 0;
found = !isset_CF();
```
Much nicer!